keps/sig-apps/0034-maxunavailable-for-statefulsets.md

```diff
@@ -0,0 +1,179 @@
+---
+kep-number: 34
+title: Implement maxUnavailable for StatefulSets
+authors:
+  - "@krmayankk"
+owning-sig: sig-apps
+participating-sigs:
+  - sig-apps
+reviewers:
+  - "@janetkuo"
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2018-12-29
+last-updated: 2018-12-29
+status: provisional
+see-also:
+  - n/a
+replaces:
+  - n/a
+superseded-by:
+  - n/a
+---
+
+# Implement maxUnavailable in StatefulSet
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [User Stories [optional]](#user-stories-optional)
+      * [Story 1](#story-1)
+    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks [optional]](#drawbacks-optional)
+* [Alternatives [optional]](#alternatives-optional)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+
+The purpose of this enhancement is to implement maxUnavailable for StatefulSet during RollingUpdate. When a StatefulSet’s 
+`.spec.updateStrategy.type` is set to `RollingUpdate`, the StatefulSet controller will delete and recreate each Pod 
+in the StatefulSet. The updating of each Pod currently happens one at a time. With support for `maxUnavailable`, the updating
+will proceed `maxUnavailable` number of pods at a time.
+
+
+## Motivation
+
+Consider the following scenarios:-
+
+1: My containers publish metrics to a time series system. If I am using a Deployment, each rolling update creates a new pod name and hence the metrics 
+published by these new pod starts a new time series which makes tracking metrics for the application difficult. While this could be mitigated, 
```

Comment:
> these new pod
Do you mean just the new pod, or collection of all old pods and the new pod?

Reply:
the old pod, has a time series based on its old hostname(pod name) going on called lets say x
the new pod, starts a new time series based on its new hostname(pod name) called lets say y